src/test/java/com/academia/library/controller/AuthControllerTest.java

```diff
@@ -0,0 +1,156 @@
+import com.academia.library.util.AuthorityName;
+import com.academia.library.util.RoleNames;
+import com.academia.library.util.TestDataCreator;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.security.crypto.password.PasswordEncoder;
+import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+
+import java.time.LocalDateTime;
+import java.util.Set;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@ActiveProfiles("test")
+@SpringBootTest
+@AutoConfigureMockMvc
+class AuthControllerTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private AuthorityRepository authorityRepository;
+
+    @Autowired
+    private RoleRepository roleRepository;
+
+    @Autowired
+    private UserRepository userRepository;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @Autowired
+    private UserMapper userMapper;
+
+    @Autowired
+    private PasswordEncoder passwordEncoder;
+
+    @Autowired
+    private UserCryptor userCryptor;
+
+    @BeforeEach
+    private void insertTestData() {
+        Authority authority = new Authority();
+        authority.setName(AuthorityName.AUTHORITY_ALL);
+        authorityRepository.save(authority);
+
+        Role role = new Role();
+        role.setName(RoleNames.USER_ROLE);
+        role.setAuthorities(Set.of(authority));
+        roleRepository.save(role);
+
+        User user = userMapper.toEntity(TestDataCreator.INSERT_USER);
+        user.setCreateAt(LocalDateTime.now());
+        user.setUpdateAt(LocalDateTime.now());
+        user.setRoles(Set.of(role));
+        user.setPassword(passwordEncoder.encode(user.getPassword()));
+        user.setEmail(userCryptor.encode(user.getEmail()));
+        user.setFirstName(userCryptor.encode(user.getFirstName()));
+        user.setLastName(userCryptor.encode(user.getLastName()));
+        userRepository.save(user);
+    }
+
+    @AfterEach
+    private void deleteAll() {
+        userRepository.deleteAll();
+        roleRepository.deleteAll();
+        authorityRepository.deleteAll();
+    }
+
+    @DisplayName("Login user")
```

Comment:
Please add this one everywhere

Reply:
Ok, 
I add this moment in Endpoint task